### PR TITLE
Revert the BOM support from 3.52 to fix incremental plugin builds + Add integration tests for Incrementals

### PR DESCRIPTION
**Pull request by jglick**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -840,140 +866,6 @@
   </distributionManagement>
 
   <profiles>
-    <profile>
-      <!--
-       The 'no-jenkins-bom' profile captures the properties & dependencies when _not_ using the Jenkins BOM:
-        https://github.com/jenkinsci/jenkins/blob/master/bom/pom.xml
-       When the BOM is not being used, this profile is adding back the <dependencyManagement /> entries that were
```
Comment (jglick) [inline pom.xml]: note merge conflict resolution: https://github.com/jenkinsci/plugin-pom/pull/229#discussion_r346328966

**Comment by jglick** [comment]
Please do not squash-merge by the way; the order of commits here is deliberate.

**Comment by jglick** [comment]
The first commit actually passes, but Jenkins was building it merged with the `master` branch, so you see the bug:

```
[INFO] Generating flattened POM of project io.jenkins.plugins:incrementals-and-plugin-bom:hpi:1.0-SNAPSHOT...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  15.019 s
[INFO] Finished at: 2019-11-14T14:07:24Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:flatten-maven-plugin:1.1.0:flatten (flatten) on project incrementals-and-plugin-bom: 8 problems were encountered while building the effective model for io.jenkins.plugins:incrementals-and-plugin-bom:1.0-SNAPSHOT
[ERROR] [ERROR] 'dependencies.dependency.version' for com.github.spotbugs:spotbugs-annotations:jar is missing. @ 
[ERROR] [ERROR] 'dependencies.dependency.version' for net.jcip:jcip-annotations:jar is missing. @ 
[ERROR] [ERROR] 'dependencies.dependency.version' for org.slf4j:slf4j-api:jar is missing. @ 
[ERROR] [ERROR] 'dependencies.dependency.version' for org.slf4j:log4j-over-slf4j:jar is missing. @ 
[ERROR] [ERROR] 'dependencies.dependency.version' for org.slf4j:jcl-over-slf4j:jar is missing. @ 
[ERROR] [ERROR] 'dependencies.dependency.version' for org.slf4j:slf4j-jdk14:jar is missing. @ 
[ERROR] [ERROR] 'dependencies.dependency.version' for org.kohsuke:access-modifier-annotation:jar is missing. @ 
[ERROR] [ERROR] 'dependencies.dependency.version' for commons-logging:commons-logging:jar is missing. @
[ERROR] -> [Help 1]
```

**Comment by jtnord** [comment]
upstream cause https://github.com/mojohaus/flatten-maven-plugin/issues/117

**Pull request merged**